File: 2017/AUG/COOK85/OBTTRNGL/OBTTRNGL.cpp
// Code made by Monal5031/monalshadi
#include "bits/stdc++.h"

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	long int tc;
	cin >> tc;
	while (tc--) {
		long long k, a, b;
		cin >> k >> a >> b;

		long long maxi = max(a, b);
		long long mini = min(a, b);

		if (maxi - mini == k/2 && k % 2 == 0) {
			cout << "0\n";
		} else if (maxi - mini - 1 < k/2) {
			cout << (maxi - mini - 1) << "\n";
		} else {
			cout << (k - maxi + mini - 1) << "\n";
		}
	}
	return 0;
}

File: 2017/AUG/COOK85/OBTTRNGL/OBTTRNGL.py
t = int(input())
for i in range(t):
    n,x,y = map(int ,input().split())
    a = min(x,y)
    b = max(x,y)
    b = b - a + 1
    a = 1
    if (n%2 == 0 and n/2 == b - a) or (b - a == 1 or b == n):
        print("0")
    else:
        if b > n/2 + 1:
            print(n - b)
        elif b < n/2 + 1:
            print(b - 2)
            